wire-library/wire-runtime-swift/src/main/swift/JSONString.swift

```diff
@@ -66,8 +68,31 @@ public struct JSONString<T : Hashable> : Hashable, Codable {
                 try container.encode(String(value))
             }
         default:
-            fatalError("Unsupported type \(T.self)")
+            if T.self is Optional<Int64>.Type || T.self is Optional<UInt64>.Type {
+                var container = encoder.singleValueContainer()
+                try container.encodeNil()
+            } else {
+                fatalError("Unsupported type \(T.self)")
+            }
         }
     }
 
 }
+
+private extension Decoder {
+    func decodeStringTo<T, Integer: FixedWidthInteger>(_ type: Integer.Type) throws -> T {
```

Comment:
Performance nit: Speed matters in Wire and generics add additional function call overhead. Since these are both pretty small and we just have two types (`Int64` and `UInt64`), can we copy/paste these methods and have a variant for both types?

I know it seems crazy, but generics definitely added to execution time in my profiling.